zip-0311.rst

```diff
@@ -2,7 +2,262 @@
+  authority [0..tx.vin.len()]
+
+  - `index`: An index into `tx.vin`.
+  - `sig`: A BIP 322 signature.
+
+    - TODO: `zcashd` currently only supports the legacy format defined in BIP 322. We may
+      want to backport full BIP 322 support before having transparent input support in
+      this ZIP, to ensure it does what we need.
+    - TODO: BIP 322 specifies consensus rule checks as part of the signature verification
+      process. We will likely need to migrate it over to an equivalent ZIP that specifies
+      these for Zcash (which has a different set of script validation consensus rules).
+
+Creating a payment disclosure
+-----------------------------
+
+The inputs to a payment disclosure are:
+
+- The transaction.
+- The SLIP-44 [#slip-0044]_ coin type.
+- The message :math:`msg` to be included (which may be empty).
+- A vector of `(output_index, ock)` tuples.
+- A vector of tuples of:
+
+  - A Sapling spend index.
+  - Its corresponding expanded spending key :math:`(\mathsf{ask}, \mathsf{nsk}, \mathsf{ovk})`.
+  - (Optional) An associated payment address :math:`(\mathsf{d}, \mathsf{pk_d})`.
+
+- A vector of tuples of:
+
+  - `index`: An index into `tx.vin`.
+  - The inputs to a BIP 322 signature (excluding `message_data`).
+
+The payment disclosure is created as follows:
+
+- For each Sapling spend index:
+
+  - Create a Sapling spend proof for the note that was spent in `tx.shieldedSpends[index]`,
+    using the same anchor, to obtain `(cv, rk, zkproof_spend)` as well as the random α
+    that was generated internally.
+
+  - [Optional] If an associated payment address was provided for this spend index, create
+    a ZIP 304 signature proof for that payment address, using α and `rk` from the
+    previous step. We obtain `(nullifier_addr, zkproof_addr)` from this step.
+
+- For each transparent input index:
+
+  - TODO: Prepare BIP 322 signature inputs.
+
+- Construct an unsigned payment disclosure from the disclosed Sapling outputs, and the
+  above data for the Sapling spends and transparent inputs.
+
+- For each Sapling spend index:
+
+  - Let :math:`\mathsf{rsk} = \mathsf{SpendAuthSig.RandomizePrivate}(α, \mathsf{ask})`.
+
+  - Let :math:`coinType` be the 4-byte little-endian encoding of the coin type in its
+    index form, not its hardened form (i.e. 133 for mainnet Zcash).
+
+  - Let :math:`digest = \mathsf{BLAKE2b}\text{-}\mathsf{256}(\texttt{"ZIP311Signed"}\,||\,coinType, payment_disclosure_digest)`.
+
+  - Let :math:`spendAuthSig = \mathsf{SpendAuthSig.Sign}(\mathsf{rsk}, digest)`.
+
+- For each transparent input index:
+
+  - TODO: Create a BIP 322 signature.
+
+- Return the payment disclosure.
+
+Verifying a payment disclosure
+------------------------------
+
+UIs displaying a payment disclosure MUST support three levels of assurance in a payment disclosure:
+
+- Level 1: The payment disclosure contains valid `ock`s that disclose the specified outputs in the specified transaction.
```

Comment:
This section no longer applies, now that we require all payment disclosures to prove spend authority.

Reply:
Give rationale for requiring that.